fabric-fluid-v1/src/main/java/net/fabricmc/fabric/api/fluid/v1/properties/FluidPropertyMerger.java

```diff
@@ -0,0 +1,59 @@
+package net.fabricmc.fabric.api.fluid.v1.properties;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.collect.Sets;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.Tag;
+
+import net.fabricmc.fabric.api.util.NbtIdentifier;
+
+public final class FluidPropertyMerger {
+	public static final FluidPropertyMerger INSTANCE = new FluidPropertyMerger();
+	private final Map<NbtIdentifier, FluidProperty<Tag>> data = new HashMap<>();
+
+	private FluidPropertyMerger() {
+	}
+
+	@SuppressWarnings ("unchecked")
+	public <T extends Tag> void register(NbtIdentifier identifier, FluidProperty<T> property) {
```

Comment:
Should we allow multiple properties with the same string keys but different types? they still can't go on a fluid together...

Reply:
how else do you resolve it, hard crash?